src/tools/concurrency/TracingBackend.java

```diff
@@ -489,10 +556,14 @@ private void processTraceData(final FileOutputStream traceDataStream,
     }
 
     @Override
+    @TruffleBoundary
```

Comment:
Why is this necessary?
This is a `run()` method on a thread, no? This should never be reached by Truffle.